lib/puppet/application/apply.rb

```diff
@@ -185,17 +185,23 @@ def main
       facts.name = Puppet[:node_name_value]
     end
 
-    configured_environment = Puppet.lookup(:current_environment)
+    # Find our Node
+    unless node = Puppet::Node.indirection.find(Puppet[:node_name_value])
+      raise "Could not find node #{Puppet[:node_name_value]}"
+    end
+
+    configured_environment = node.environment || Puppet.lookup(:current_environment)
+
     apply_environment = manifest ?
       configured_environment.override_with(:manifest => manifest) :
       configured_environment
 
-    Puppet.override({:current_environment => apply_environment}, "For puppet apply") do
-      # Find our Node
-      unless node = Puppet::Node.indirection.find(Puppet[:node_name_value])
-        raise "Could not find node #{Puppet[:node_name_value]}"
-      end
+    # Make the node use the its usual environment, with :manifest overridden
+    # to the manifest from the command line. If the node terminus returned no
+    # environment, the locally configured environment was overridden instead.
```

Comment:
On line 193, we do:

<pre>
configured_environment = node.environment || Puppet.lookup(:current_environment)
</pre>

which seems to contradict the comment. If the node terminus returns no environment, then we do *not* override the locally configured environment, right?